File: По-сложни проверки/on_time_for_the_exam.py

exam_hour = int(input())
exam_min = int(input())
arrived_hour = int(input())
arrived_min = int(input())

exam_time = (exam_hour * 60) + exam_min
arrived_time = (arrived_hour * 60) + arrived_min
result = abs(exam_time - arrived_time)

if exam_time < arrived_time: #late
    converted_hour = result // 60
    converted_min = result % 60
    print("Late")
    if converted_hour == 0:
        print(f"{converted_min} minutes after the start")
    else:
        print(f"{converted_hour}:{converted_min:02d} hours after the start")
else: #on time OR early
    if result <= 30: #on time
        converted_min = result % 60
        print("On time")
        print(f"{converted_min} minutes before the start")
    else: #early
        converted_hour = result // 60
        converted_min = result % 60
        print("Early")
        if converted_hour == 0:
           print(f"{converted_min} minutes before the start") 
        else:
            print(f"{converted_hour}:{converted_min:02d} hours before the start" )

File: Programming Basics with Java/More complex checks/OnTimeForTheExam.java
import java.util.Scanner;

public class OnTimeForTheExam 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        int test_hour = input.nextInt();
        int test_min = input.nextInt();
        int arrival_hour = input.nextInt();
        int arrival_min = input.nextInt();

        test_min += test_hour * 60;
        arrival_min += arrival_hour * 60;

        int diff = test_min - arrival_min;
        int diff_hour = Math.abs(diff) / 60;
        int diff_min = Math.abs(diff) % 60;

        if (diff < 0)
        {
            System.out.println("Late");
            if (diff_hour != 0)
            {
                System.out.printf("%d:%02d hours after the start", diff_hour, diff_min);
            }
            else
            {
                System.out.printf("%02d minutes after the start", diff_min);
            }
        }
        else if (diff > 30)
        {
            System.out.println("Early");
            if (diff_hour != 0)
            {
                System.out.printf("%d:%02d hours before the start", diff_hour, diff_min);
            }
            else
            {
                System.out.printf("%02d minutes before the start", diff_min);
            }
        }
        else
        {
            System.out.println("On time");
            if (diff != 0)
            {
                System.out.printf("%d minutes before the start", diff);
            }
        }
    }
}
